lib/Rename/Adapter/ClassMover/FileRenamer.php

```diff
@@ -2,90 +2,79 @@
 
 namespace Phpactor\Rename\Adapter\ClassMover;
 
-use Amp\Promise;
+use Generator;
 use Phpactor\ClassMover\ClassMover;
 use Phpactor\Indexer\Model\QueryClient;
 use Phpactor\Rename\Model\Exception\CouldNotConvertUriToClass;
 use Phpactor\Rename\Model\Exception\CouldNotRename;
 use Phpactor\Rename\Model\FileRenamer as PhpactorFileRenamer;
 use Phpactor\Rename\Model\LocatedTextEdit;
-use Phpactor\Rename\Model\LocatedTextEditsMap;
+use Phpactor\Rename\Model\RenameResult;
 use Phpactor\Rename\Model\UriToNameConverter;
 use Phpactor\TextDocument\Exception\TextDocumentNotFound;
 use Phpactor\TextDocument\TextDocumentLocator;
 use Phpactor\TextDocument\TextDocumentUri;
-use Phpactor\TextDocument\TextEdits;
-use function Amp\call;
 
 class FileRenamer implements PhpactorFileRenamer
 {
     public function __construct(
         private UriToNameConverter $converter,
         private TextDocumentLocator $locator,
         private QueryClient $client,
-        private ClassMover $mover
+        private ClassMover $mover,
     ) {
     }
 
-    /**
-     * @return Promise<LocatedTextEditsMap>
-     */
-    public function renameFile(TextDocumentUri $from, TextDocumentUri $to): Promise
+    public function renameFile(TextDocumentUri $from, TextDocumentUri $to): Generator
```

Comment:
I switched to using generator in similar way like in `ClassRenamer` (please compare what happens there) to smuggle `RenameResult` through `getReturn()` and avoid changes in models (as I mentioned above).

Reply:
Using Promise would require some new class (`Phpactor\Rename\Model\RenameEdit`?) composed of `LocatedTextEditsMap` and `RenameResult`.

Reply:
`RenameEdit` class couldn't simply look like
```php
<?php

namespace Phpactor\Rename\Model;

class RenameEdit
{
    public function __construct(
        public readonly LocatedTextEditsMap $locatedTextEditsMap,
        public readonly RenameResult $renameResult,
    ) {
    } 
}
```
because it'd loss the order of edits. Such class should be similar to `WorkspaceEdit` that allows ordered edits of different types.

Reply:
I converted the adapter to return Promise and created `RenameEdit` - initially as iterable class.